File: report-generator/convert_svg.py
import cairosvg
import os

# 转换SVG为PNG
def convert_svg_to_png(svg_path, png_path):
    try:
        cairosvg.svg2png(url=svg_path, write_to=png_path, output_width=800, output_height=600)
        print(f"转换成功：{png_path}")
        return True
    except Exception as e:
        print(f"转换失败：{e}")
        return False

# 主函数
def main():
    # 图表路径
    charts = [
        ('temp/architecture.svg', 'temp/architecture.png'),
        ('temp/module.svg', 'temp/module.png'),
        ('temp/er.svg', 'temp/er.png'),
        ('temp/user-entity.svg', 'temp/user-entity.png')
    ]
    
    # 转换所有图表
    for svg_path, png_path in charts:
        if os.path.exists(svg_path):
            convert_svg_to_png(svg_path, png_path)
        else:
            print(f"文件不存在：{svg_path}")

if __name__ == "__main__":
    main()

File: report-generator/convert-svg.js
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

// 转换SVG为PNG
async function convertSvgToPng(svgPath, pngPath) {
  try {
    const svgBuffer = fs.readFileSync(svgPath);
    await sharp(svgBuffer)
      .resize(800, 600)
      .png()
      .toFile(pngPath);
    console.log(`转换成功：${pngPath}`);
    return true;
  } catch (error) {
    console.error(`转换失败：${error.message}`);
    return false;
  }
}

// 主函数
async function main() {
  // 图表路径
  const charts = [
    ['temp/architecture.svg', 'temp/architecture.png'],
    ['temp/module.svg', 'temp/module.png'],
    ['temp/er.svg', 'temp/er.png'],
    ['temp/user-entity.svg', 'temp/user-entity.png']
  ];
  
  // 转换所有图表
  for (const [svgPath, pngPath] of charts) {
    if (fs.existsSync(svgPath)) {
      await convertSvgToPng(svgPath, pngPath);
    } else {
      console.log(`文件不存在：${svgPath}`);
    }
  }
}

// 运行主函数
main().catch(console.error);